src/main/java/com/google/error_analyzer/backend/MockLogDao.java

```diff
@@ -0,0 +1,112 @@
+/**Copyright 2019 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.*/
+
+package com.google.error_analyzer.backend;
+
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.lang.*;
+import java.util.*;
+import org.elasticsearch.common.text.Text;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.fetch.subphase.highlight.HighlightField;
+
+public class MockLogDao implements DaoInterface {
+    private final String[] database = new String[] {"Error: nullPointerException", 
+    "info: start appengine","scheduler shutting down",
+    "WARNING: An illegal reflective access operation has occurred", 
+    "Severe: Could not find index file", "warning: NullPointerException"};
+
+    //search db using keywords and return searchHits having highlight field added 
+    @Override 
+    public ImmutableList < SearchHit > fullTextSearch
+    (String fileName, String searchString, String field) throws IOException 
+    {
+        ArrayList < SearchHit > searchResults = new ArrayList();
```

Comment:
There isn't a need to create searchResults of type ArrayList.
Instead initialise the ImmutableList builder here.

Add the elements to the builder in for loop.

build() it in the end.